### PR TITLE
[MAINTENANCE] New PR template

### PR DESCRIPTION
**Pull request by kenwade4**
Changes proposed in this pull request:
- New PR template

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,32 +1,17 @@
...
+- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
+- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
+- [ ] Code is linted
 
-For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.
+    ```
+    black .
 
-In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).
+    ruff . --fix
```
Comment (cdkini) [inline .github/pull_request_template.md:11]: How about `invoke lint`?

Comment (cdkini) [reply .github/pull_request_template.md:11]:
```
- [ ] Code is linted (run `invoke lint`)
```
or something to that effect

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,32 +1,17 @@
...
+    ```
+- [ ] Appropriate tests and docs have been updated
 
-Previous Design Review notes:
--
--
--
+For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).
```
Comment (cdkini) [inline .github/pull_request_template.md:15]: Could we make the capitalization of these hyperlinks consistent?

File: .github/pull_request_template.md (hunk 1 above)
Comment (Kilo59) [reply .github/pull_request_template.md:11]: `invoke lint` only runs `ruff` not the `black` formatting.

For historical reasons `invoke fmt` runs black & just the ruff `isort` rules. 

We could change some of this behavior if we wanted `invoke lint` to format and run `ruff`.

**Review by Kilo59 — APPROVED** [review]
I like it!

File: .github/pull_request_template.md (hunk 2 above)
Comment (kenwade4) [reply .github/pull_request_template.md:15]: Sure. Rather make Checklist to checklist?

File: .github/pull_request_template.md (hunk 1 above)
Comment (kenwade4) [reply .github/pull_request_template.md:11]: Just being consistent with https://docs.greatexpectations.io/docs/contributing/style_guides/code_style/#linting and other places like https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_aggregate_expectations/#8-linting

Comment (kenwade4) [reply .github/pull_request_template.md:11]: Want fewer things confusing users. Sometimes invoke tasks lead to me needing more troubleshooting, but the lint task usually isn't one of them.

**Review by anthonyburdi — COMMENTED** [review]
+1 (pending resolving Chetan's comments). Thank you for making these changes!

File: .github/pull_request_template.md (hunk 2 above)
Comment (kenwade4) [reply .github/pull_request_template.md:15]: @cdkini do you have a preference?

**Review by anthonyburdi — APPROVED** [review]
LGTM! This is much clearer.

File: .github/pull_request_template.md (hunk 1 above)
Comment (anthonyburdi) [reply .github/pull_request_template.md:11]: What about `invoke fix`?

**Pull request merged**